modules/azure-eventhub/variables.tf

```
@@ -0,0 +1,130 @@
+variable "location" {
+  description = "Azure region where the Event Hub namespace is deployed."
+  type        = string
+}
+
+variable "resource_group_name" {
+  description = "Name of the resource group that will contain the Event Hub namespace."
+  type        = string
+}
+
+variable "tags" {
+  description = "Tags applied to all resources created by this module."
+  type        = map(string)
+  default     = {}
+}
+
+variable "namespace" {
+  description = "Configuration for the Event Hub namespace."
+  type = object({
+    name                          = string
+    sku                           = optional(string, "Standard")
+    capacity                      = optional(number)
+    auto_inflate_enabled          = optional(bool, false)
+    maximum_throughput_units      = optional(number)
+    minimum_tls_version           = optional(string, "1.2")
+    public_network_access_enabled = optional(bool, true)
+    local_authentication_enabled  = optional(bool, true)
+    identity = optional(object({
+      type         = string
+      identity_ids = optional(list(string), [])
+    }))
+    customer_managed_key = optional(object({
+      key_vault_key_ids         = list(string)
+      user_assigned_identity_id = optional(string)
+    }))
+    network_rules = optional(object({
+      default_action                 = optional(string, "Deny")
+      trusted_service_access_enabled = optional(bool, false)
+      public_network_access_enabled  = optional(bool, true)
+      ip_rules = optional(list(object({
+        ip_mask = string
+        action  = optional(string, "Allow")
+      })), [])
+      virtual_network_rules = optional(list(object({
+        subnet_id                                       = string
+        ignore_missing_virtual_network_service_endpoint = optional(bool, false)
+      })), [])
+    }))
+    create_listen_rule = optional(bool, true)
+    create_send_rule   = optional(bool, true)
+    create_manage_rule = optional(bool, false)
+    tags               = optional(map(string), {})
```

```suggestion
    listen_rule_enabled = optional(bool, true)
    send_rule _enabled  = optional(bool, true)
    manage_rule_enabled = optional(bool, false)
    extra_tags               = optional(map(string), {})
```

extra because it signals you merge